Homework2/Program.cs

```diff
@@ -0,0 +1,171 @@
+
+            SignUp();
+            
+        }
+
+        static int[] SortArray(int[] arr)
+        {
+            int[] copiedArray = new int[arr.Length];
+            Array.Copy(arr, copiedArray, arr.Length);
+            for (int i = 0; i < copiedArray.Length - 1; i++)
+            {
+                int j;
+                int minimum = copiedArray[i];
+                int minIndex = i;
+                for (j = i + 1; j < copiedArray.Length; j++)
+                {
+                    if (copiedArray[j] < minimum)
+                    {
+                        minimum = copiedArray[j];
+                        minIndex = j;
+                    }
+                }
+                int temp = copiedArray[i];
+                copiedArray[i] = minimum;
+                copiedArray[minIndex] = temp;
+            }
+            return copiedArray;
+        }
+
+        static int[] AddToStart(int[] arr, int item)
+        {
+            int[] copiedArray = new int[arr.Length + 1];
+            Array.Copy(arr, 0, copiedArray, 1, arr.Length);
+            copiedArray[0] = item;
+            return copiedArray;
+        }
+
+        static int[] AddToEnd(int[] arr, int item)
+        {
+            int[] copiedArray = new int[arr.Length + 1];
+            Array.Copy(arr, copiedArray, arr.Length);
+            copiedArray[copiedArray.Length - 1] = item;
+            return copiedArray;
+        }
+
+        static int[] AddToArray(int[] arr, int item, int index)
+        {
+            if (index == 0) return AddToStart(arr, item);
+            else if (index == arr.Length - 1) return AddToEnd(arr, item);
+            int[] copiedArray = new int[arr.Length + 1];
+            Array.Copy(arr, copiedArray, index);
+            copiedArray[index] = item;
+            Array.Copy(arr, index, copiedArray, index + 1, arr.Length - index);
+            return copiedArray;
+        }
+
+        static int[] RemoveFromStart(int[] arr)
+        {
+            int[] copiedArray = new int[arr.Length - 1];
+            Array.Copy(arr, 1, copiedArray, 0, copiedArray.Length);
+            return copiedArray;
+        }
+
+        static int[] RemoveFromEnd(int[] arr)
+        {
+            int[] copiedArray = new int[arr.Length - 1];
+            Array.Copy(arr, copiedArray, copiedArray.Length);
+            return copiedArray;
+        }
+
+        static int[] RemoveFromArray(int[] arr, int index)
+        {
+            if (index == 0) return RemoveFromStart(arr);
+            else if (index == arr.Length - 1) return RemoveFromEnd(arr);
+            int[] copiedArray = new int[arr.Length - 1];
+            Array.Copy(arr, copiedArray, index);
+            Array.Copy(arr, index + 1, copiedArray, index, copiedArray.Length - index);
+            return copiedArray;
+        }
+
+        static void SignUp()
+        {
+            Console.Write("Add an username:");
```

Comment:
Technically "a username".  I wouldn't pick on this normally but I'm finding very little wrong so I have to nitpick where I can :)